### PR TITLE
CPS-151: v1.24.0 aws credentials file handling

### PR DESCRIPTION
**Pull request by Johnbastian**
[CPS-149]
[CPS-141]

# Main change
The file handling will now perform two things in development mode:
- the target bucket will be `workflow-file-uploads-dev`
- on startup, `aws.json` usage in the entry script which invokes falafel (`main.js`) is checked and a console warning will be shown

## Other changes
- Directories in the helpers folder which do not have an `index.js` will now be ignored instead of erring.
- Documentation has been added for file handling

[CPS-149]: https://trayio.atlassian.net/browse/CPS-149
[CPS-141]: https://trayio.atlassian.net/browse/CPS-141

### CODE REVIEW

File: lib/fileHandler/index.js [hunk 1]
```diff
@@ -1,23 +1,25 @@
...
+const needle = require('needle');
 
-var AWS = require('aws-sdk');
+const AWS = require('aws-sdk');
 
-var REGION = 'us-west-2',
-	BUCKET = 'workflow-file-uploads';
+const REGION = 'us-west-2',
+	BUCKET = 'workflow-file-uploads',
+	DEV_BUCKET = 'workflow-file-uploads-dev';
```
Comment (adamtarrant) [inline lib/fileHandler/index.js]: Should we consider creating our `workflow-file-uploads-dev` bucket in the same region as our dev environment (eu-west-2)? Even though I think the AWS console shows all regions, buckets do have a region setting I believe.

Comment (PiotrTrzpil) [reply lib/fileHandler/index.js]: yeah could be in eu-west-2

Comment (Johnbastian) [reply lib/fileHandler/index.js]: Hmm, just tried setting the region to `eu-west-2`, but even though it worked, the signed URL didn't look like it normally does. Also, the URL still points to `us-west-2`.

Looking at the config Luka setup for this bucket, seems like it's configured for `us-west-2` - so will keep it as is (for now; can change later if needed).

File: lib/fileHandler/index.js [hunk 2]
```diff
@@ -27,20 +29,31 @@ function logError (errorName, error) {
...
 
+	const TARGET_BUCKET = ( options.dev || options.test ? DEV_BUCKET : BUCKET );
+
 	//Pull the AWS creds from environment if present
 	if (_.isPlainObject(options.aws)) {
-		var specifiedCreds = options.aws;
-		if (specifiedCreds.key && specifiedCreds.secret) {
+		if (options.dev) {
+			// eslint-disable-next-line no-console
+			console.warn('Specifying `key` and `secret` is not recommended. Please consider setting up AWS credentials or another method supported by AWS SDK instead.');
```
[the remark this answers is not in the document]
Comment (Johnbastian) [reply lib/fileHandler/index.js]: a) Yes, tstst does setup environment variable, and that would still be accessible by the SDK. The `key` and `secret` referred to here are values manually passed in. If you check what the `aws.json` file was exporting, that's what was being passed around as `options.aws`.
b) I already asked Luka, and connector development is not a use case for tstst - so don't try using that. I've prepared some docs to instruct how develop file handling connectors without using `aws.json`.

**Pull request merged**